File: common/src/main/java/org/hiero/mirror/common/domain/contract/ContractTransactionHash.java
// SPDX-License-Identifier: Apache-2.0

package org.hiero.mirror.common.domain.contract;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.domain.Persistable;

@Builder
@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class ContractTransactionHash implements Persistable<byte[]> {
    private long consensusTimestamp;

    private long entityId;

    @Id
    private byte[] hash;

    private long payerAccountId;

    private Integer transactionResult;

    @JsonIgnore
    @Override
    public byte[] getId() {
        return hash;
    }

    @JsonIgnore
    @Override
    public boolean isNew() {
        return true; // Since we never update and use a natural ID, avoid Hibernate querying before insert
    }
}

File: rest/model/contractTransactionHash.js
// SPDX-License-Identifier: Apache-2.0

class ContractTransactionHash {
  static CONSENSUS_TIMESTAMP = 'consensus_timestamp';
  static ENTITY_ID = 'entity_id';
  static HASH = 'hash';
  static PAYER_ACCOUNT_ID = 'payer_account_id';
  static TRANSACTION_RESULT = 'transaction_result';
  static tableName = 'contract_transaction_hash';

  /**
   * Parses contract_transaction_hash table columns into object
   */
  constructor(transactionHash) {
    this.consensusTimestamp = transactionHash[ContractTransactionHash.CONSENSUS_TIMESTAMP];
    this.entityId = transactionHash[ContractTransactionHash.ENTITY_ID];
    this.hash = transactionHash[ContractTransactionHash.HASH];
    this.payerAccountId = transactionHash[ContractTransactionHash.PAYER_ACCOUNT_ID];
    this.transactionResult = transactionHash[ContractTransactionHash.TRANSACTION_RESULT];
  }
}

export default ContractTransactionHash;
